unit_tests/testExtendedMath/testExtendedMath.f90

```diff
@@ -81,67 +76,66 @@ SUBROUTINE testRatFrac()
     tol=10.0_SRK**(-REAL(i,SRK))
     CALL Rational_Fraction(SQRT(2.0_SRK),tol,N,D)
     dif=ABS(REAL(N,SRK)/REAL(D,SRK)/SQRT(2.0_SRK) - 1.0_SRK)
-    ASSERT(dif<=tol, 'tolerance check')
-    FINFO() i, N, D, ABS(SQRT(2.0_SRK)-REAL(N,SRK)/REAL(D,SRK)), tol
+    ASSERT_LE(dif,tol, 'tolerance check')
   ENDDO
 
 ENDSUBROUTINE testRatFrac
 !
 !-------------------------------------------------------------------------------
 SUBROUTINE testGCD()
   ! test error conditions (passing negative or 0 values)
-  ASSERT(GreatestCommonDivisor(-1,5) == 0, 'GCD errors')
-  ASSERT(GreatestCommonDivisor(5,-5) == 0, 'GCD errors')
-  ASSERT(GreatestCommonDivisor(-1,-5) == 0, 'GCD errors')
-  ASSERT(GreatestCommonDivisor(0,5) == 0, 'GCD errors')
-  ASSERT(GreatestCommonDivisor(5,0) == 0, 'GCD errors')
-
-  ASSERT(GreatestCommonDivisor(2,3) == 1, 'GCD 2,3')
-  ASSERT(GreatestCommonDivisor(3,2) == 1, 'GCD 3,2')  ! ensure both directions give same answer
-  ASSERT(GreatestCommonDivisor(201,3) == 3, 'GCD 201,3')
-  ASSERT(GreatestCommonDivisor(33,198) == 33, 'GCD 33,198')
-  ASSERT(GreatestCommonDivisor(6248,3784) == 88, 'GCD 6248,3784')
+  ASSERT_EQ(GreatestCommonDivisor(-1,5),0, 'GCD errors')
+  ASSERT_EQ(GreatestCommonDivisor(5,-5),0, 'GCD errors')
+  ASSERT_EQ(GreatestCommonDivisor(-1,-5),0, 'GCD errors')
+  ASSERT_EQ(GreatestCommonDivisor(0,5),0, 'GCD errors')
+  ASSERT_EQ(GreatestCommonDivisor(5,0),0, 'GCD errors')
+
+  ASSERT_EQ(GreatestCommonDivisor(2,3),1, 'GCD 2,3')
+  ASSERT_EQ(GreatestCommonDivisor(3,2),1, 'GCD 3,2')  ! ensure both directions give same answer
+  ASSERT_EQ(GreatestCommonDivisor(201,3),3, 'GCD 201,3')
+  ASSERT_EQ(GreatestCommonDivisor(33,198),33, 'GCD 33,198')
+  ASSERT_EQ(GreatestCommonDivisor(6248,3784),88, 'GCD 6248,3784')
 ENDSUBROUTINE testGCD
 !
 !-------------------------------------------------------------------------------
+SUBROUTINE testLCM()
+  ASSERT_EQ(LeastCommonMultiple(-1,5),5, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(5,-5),5, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(-1,-5),5, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(0,5),0, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(5,0),0, 'LCM')
+
+  ASSERT_EQ(LeastCommonMultiple(2,3),6, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(3,2),6, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(201,3),201, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(33,198),198, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(8,12),24, 'LCM')
+  ASSERT_EQ(LeastCommonMultiple(8,36),72, 'LCM')
+
+  ASSERT_EQ(LeastCommonMultiple((/8,12,36/)),LeastCommonMultiple(8,36),"LCM array")
```

Comment:
Might be good to have a couple tests with longer arrays and things in non-ascending orders and such just to make sure nothing crazy happens.  But overall the test is in pretty decent shape

Reply:
Added a few more plus a couple more "wonky" cases.